restli-server/src/main/java/com/linkedin/restli/internal/server/methods/arguments/ArgumentBuilder.java

```diff
@@ -482,12 +493,11 @@ private static Object buildRegularArgument(final ResourceContext context,
     return convertedValue;
   }
 
-  private static DataTemplate<?> buildDataTemplateArgument(final ResourceContext context,
+  private static DataTemplate<?> buildDataTemplateArgument(final Object paramValue,
                                                            final Parameter<?> param,
-                                                           final boolean validateRequestQueryParam)
+                                                           final boolean validateRequest)
```

Comment:
Just curious, why the signature change? Isn't this validation actually only apply to QueryParameters?

Reply:
this is now used for key params also which will be in the path.

Reply:
I saw your flag only guarding the following line(the result), which is supposed to validate parameter, right?
```
      DataSchemaAnnotationValidator validator = new DataSchemaAnnotationValidator(paramRecordTemplate.schema());
      ValidationResult result =
          ValidateDataAgainstSchema.validate(paramRecordTemplate.data(), paramRecordTemplate.schema(),
              new ValidationOptions(RequiredMode.CAN_BE_ABSENT_IF_HAS_DEFAULT, CoercionMode.STRING_TO_PRIMITIVE),
              validator);
```

Reply:
Oh I see, your add the line 124-132.

I see. Then I think "validateParams" might be clearer than "validateRequest"